### PR TITLE
chore: make all packages build to be consistent and compatible with cjs and esm

### PR DESCRIPTION
**Pull request by nazarhussain**
**Motivation**

Make all packages build to be consistent and compatible with cjs and esm

**Description**

- Modern tooling required the source code to be in ESM 
- Some of our packages were ESM and some were configured as commonjs. 
- The library build was also inconsistent across all packages in this repo. 
- This PR will configure the repo and all packages to be ESM modules
- Also build CJS and ESM separately and constantly for all packages

**Steps to test or reproduce**

Run all tests

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -1,7 +1,20 @@
 {
-  "extends": "./tsconfig.build.json",
   "compilerOptions": {
-    "emitDeclarationOnly": false,
-    "incremental": false,
+    "module": "NodeNext",
+    "target": "es2022",
+    "strict": true,
+    "esModuleInterop": true,
+    "skipLibCheck": true,
```
[the remark this answers is not in the document]
Comment (nazarhussain) [reply tsconfig.json:7]: Some libraries are old and don't have valid types.

File: packages/ssz/src/type/container.ts [hunk 2]
```diff
@@ -551,7 +551,7 @@ export function precomputeJsonKey<Fields extends Record<string, Type<unknown>>>(
   if (casingMap) {
     const keyFromCaseMap = casingMap[fieldName];
     if (keyFromCaseMap === undefined) {
-      throw Error(`casingMap[${fieldName}] not defined`);
+      throw Error(`casingMap[${String(fieldName as Symbol)}] not defined`);
```
[the remark this answers is not in the document]
Comment (nazarhussain) [reply packages/ssz/src/type/container.ts:554]: With latest TS upgrade it says compilation error casting error using a symbol inside string interpolation.

File: packages/as-sha256/package.json [hunk 3]
```diff
@@ -31,17 +24,19 @@
...
+    "build:esm": "tsc -p tsconfig.build.esm.json && echo '{\"type\": \"module\"}' > ./lib/esm/package.json",
+    "build:cjs": "tsc -p tsconfig.build.cjs.json && echo '{\"type\": \"commonjs\"}' > ./lib/cjs/package.json",
+    "build:types": "tsc -p tsconfig.build.types.json",
     "build:web": "webpack --mode production --entry ./index.js --output ./dist/as-sha256.min.js",
     "test": "yarn run test:unit",
-    "test:unit": "yarn run test:unit:node && yarn run test:unit:browser",
-    "test:unit:node": "mocha -r ts-node/register test/unit/*.test.ts",
-    "test:unit:browser": "karma start --single-run --browsers ChromeHeadless,FirefoxHeadless karma.config.js",
-    "benchmark": "node -r ts-node/register ./node_modules/.bin/benchmark 'test/perf/*.test.ts'",
+    "test:unit": "yarn run test:unit:node",
```
Comment (nazarhussain) [inline packages/as-sha256/package.json:36]: Disabling the browser tests temporarily, as configuring Karma with ESM modules and TS is very time consuming. And on other branch we already have migrated these to Vitest.

**Review by matthewkeil — APPROVED** [review]
Hooray!!!! LGTM 🚀

File: packages/simpleserialize.com/src/components/Input.tsx [hunk 4]
```diff
@@ -33,7 +33,7 @@ type State = {
   userHasEditedInput: boolean;
 };
 
-const DEFAULT_FORK = "deneb";
+const DEFAULT_FORK = "deneb/index.js";
```
Comment (nflaig) [inline packages/simpleserialize.com/src/components/Input.tsx:36]: @nazarhussain do you remember why this change was required? 

it breaks the website
```
index.tsx:17 TypeError: Cannot convert undefined or null to object
    at Object.keys (<anonymous>)
    at typeNames (types.ts:35:17)
    at Input.names (Input.tsx:101:12)
    at Input.render (Input.tsx:284:27)
```

I don't see any issues when reverting this to previous code, but might be missing something
